server/container_create.go

```diff
@@ -110,17 +100,16 @@ func (s *Server) CreateContainer(ctx context.Context, req *pb.CreateContainerReq
 	return resp, nil
 }
 
-func (s *Server) createSandboxContainer(containerID string, containerName string, sb *sandbox, containerDir string, containerConfig *pb.ContainerConfig) (*oci.Container, error) {
+func (s *Server) createSandboxContainer(ctx context.Context, containerID string, containerName string, sb *sandbox, SandboxConfig *pb.PodSandboxConfig, containerConfig *pb.ContainerConfig) (*oci.Container, error) {
 	if sb == nil {
 		return nil, errors.New("createSandboxContainer needs a sandbox")
 	}
+
+	// TODO: factor generating/updating the spec into something other projects can vendor
+
 	// creates a spec Generator with the default spec.
 	specgen := generate.New()
 
-	// by default, the root path is an empty string.
-	// here set it to be "rootfs".
-	specgen.SetRootPath("rootfs")
-
 	args := containerConfig.GetArgs()
 	if args == nil {
 		args = []string{"/bin/sh"}
```

Comment:
@nalind you should fix this by setting the args coming from the image's config if the `containerConfig.GetArgs()` is `nil`. Otherwise, unless you run `kubectl run nginx --image=nginx -- <arg1> <arg2> ... <argN>` you just get a shell instead of running nginx.